src/core/Do.Architecture/Architecture/LayerBase.cs

```diff
@@ -4,6 +4,8 @@ public abstract class LayerBase : ILayer
 {
     protected ApplicationContext Context { get; private set; } = default!;
 
+    public string Id => GetType().Name;
```

Comment:
implement `Id` explicitly just like the others, and provide `protected virtual string Id => GetType().Name`